app/main.py

```diff
@@ -1,34 +1,128 @@
+from __future__ import annotations
+
+
+class ErrorNumberOfTheShips(ValueError):
+    pass
+
+
+class ErrorNumberOfShipsDeck(ValueError):
+    pass
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+
+    @staticmethod
+    def decks_maker(start: tuple, end: tuple) -> list[Deck]:
+        decks = []
+        if start[0] - end[0] != 0:
+            for row in range(start[0], end[0] + 1):
+                decks.append(Deck(row, end[1]))
+        elif start[1] - end[1] != 0:
+            for column in range(start[1], end[1] + 1):
+                decks.append(Deck(start[0], column))
+        elif start == end:
+            decks.append(Deck(start[0], start[1]))
```

Comment:
u can use here only 2 conditions, try to solve this

Reply:
Fixed